### PR TITLE
Include bundle by default

### PR DESCRIPTION
**Pull request by islean**
## Description

Addresses issue #136 and adds an exclude flag to the add command. If not used, the add bundle command will include the bundle.

### Added

- Flag to exclude bundle when running `housekeeper add bundle`.

### Changed

- By default `housekeeper add bundle` includes the bundle.

### How to prepare for test

- [x] ssh to hasta (depending on type of change)
- [x] install on stage:
```bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t housekeeper -b include-bundle-by-default```

### How to test
- [x] Add a bundle
- [x] Add a file to the bundle

### Expected test outcome
- [x] Check that the bundle was added and included.
- [x] Check that adding the file works without running `include bundle`.
- [x] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] code approved by
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by islean** [comment]
Tests went well:

image

And now `add file` runs without needing to include the bundle beforehand.
image

File: housekeeper/cli/add.py [hunk 1]
```diff
@@ -70,6 +72,13 @@ def bundle_cmd(context: click.Context, bundle_name: str, json: str):
     store.session.add(new_bundle)
     new_version.bundle: Bundle = new_bundle
     store.session.add(new_version)
+    if not exclude:
+        try:
+            include_version(context.obj[ROOT], new_version)
+        except VersionIncludedError as error:
+            LOG.warning(error.message)
```
Comment (diitaz93) [inline housekeeper/cli/add.py:79]: Im thinking this should be an error message
```suggestion
            LOG.error(error.message)
```

**Review by diitaz93 — APPROVED** [review]
Looks good!

File: housekeeper/cli/add.py (hunk 1 above)
Comment (islean) [reply housekeeper/cli/add.py:79]: Open for it either way! Now I mirrored how the include command looks.

**Review by henrikstranneheim — APPROVED** [review]
👍

File: housekeeper/cli/add.py [hunk 2]
```diff
@@ -70,6 +72,13 @@ def bundle_cmd(context: click.Context, bundle_name: str, json: str):
     store.session.add(new_bundle)
     new_version.bundle: Bundle = new_bundle
     store.session.add(new_version)
+    if not exclude:
+        try:
+            include_version(context.obj[ROOT], new_version)
```
Comment (seallard) [inline housekeeper/cli/add.py:77]: What is `context.obj[ROOT]`? I found this confusing, maybe it can be made clearer somehow

**Review by seallard — APPROVED** [review]
Looks good 👍 There might be too much going on in the cli command but out of scope to fix in this PR (a lot of logic could probably be extracted from the cli)

File: housekeeper/cli/add.py (hunk 1 above)
Comment (islean) [reply housekeeper/cli/add.py]: Changed to Log.error

File: housekeeper/cli/add.py (hunk 2 above)
Comment (islean) [reply housekeeper/cli/add.py]: Clarified with variable instantiation and named arguments.

**Comment by islean** [comment]
Tests still successful with comments addressed:

```
θ67° [isak.ohlsson@hasta:/home/proj/stage] [S_base] 56s $ housekeeper add bundle TestBundle
2023-09-11 13:26:59 hasta.scilifelab.se housekeeper.cli.core[259560] INFO Use root path /home/proj/stage/housekeeper-bundles
2023-09-11 13:26:59 hasta.scilifelab.se housekeeper.cli.add[259560] INFO Running add bundle
2023-09-11 13:26:59 hasta.scilifelab.se housekeeper.store.api.handlers.read[259560] INFO Fetching bundle with name: TestBundle
2023-09-11 13:26:59 hasta.scilifelab.se housekeeper.files[259560] INFO Validating bundle schema
2023-09-11 13:26:59 hasta.scilifelab.se housekeeper.files[259560] INFO Validate marshmallow schema
2023-09-11 13:26:59 hasta.scilifelab.se housekeeper.files[259560] INFO Input looks fine
2023-09-11 13:26:59 hasta.scilifelab.se housekeeper.date[259560] INFO Trying to parse date string 2023-09-11 13:26:59.775662
2023-09-11 13:26:59 hasta.scilifelab.se housekeeper.store.api.handlers.read[259560] INFO Fetching bundle with name: TestBundle
2023-09-11 13:26:59 hasta.scilifelab.se housekeeper.store.api.handlers.create[259560] INFO Created new bundle: TestBundle
2023-09-11 13:26:59 hasta.scilifelab.se housekeeper.store.api.handlers.create[259560] INFO Created new version
2023-09-11 13:26:59 hasta.scilifelab.se housekeeper.include[259560] INFO Use global root path /home/proj/stage/housekeeper-bundles
2023-09-11 13:26:59 hasta.scilifelab.se housekeeper.include[259560] INFO created new bundle version dir: /home/proj/stage/housekeeper-bundles/TestBundle/2023-09-11
2023-09-11 13:26:59 hasta.scilifelab.se housekeeper.cli.add[259560] INFO new bundle added: TestBundle (150251)
θ65° [isak.ohlsson@hasta:/home/proj/stage] [S_base] 2s $ housekeeper add file -b TestBundle test_file.txt 
2023-09-11 13:27:22 hasta.scilifelab.se housekeeper.cli.core[8077] INFO Use root path /home/proj/stage/housekeeper-bundles
2023-09-11 13:27:23 hasta.scilifelab.se housekeeper.cli.add[8077] INFO Running add file
2023-09-11 13:27:23 hasta.scilifelab.se housekeeper.store.api.handlers.read[8077] INFO Fetching bundle with name: TestBundle
2023-09-11 13:27:24 hasta.scilifelab.se housekeeper.include[8077] INFO linked file: test_file.txt -> /home/proj/stage/housekeeper-bundles/TestBundle/2023-09-11/test_file.txt
2023-09-11 13:27:24 hasta.scilifelab.se housekeeper.cli.add[8077] INFO new file added: TestBundle/2023-09-11/test_file.txt (4483606)
```

**Comment by islean** [comment]
Deployed to stage 2023-09-11 by IO:

```
θ64° [isak.ohlsson@hasta:/home/proj/stage/housekeeper-bundles] [S_base] $ bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t housekeeper -b master

...

repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/proj/stage/housekeeper-bundles
done.
Log deploy... done.
housekeeper, version 4.8.0
```

**Comment by islean** [comment]
Deployed to production 2023-09-11 by IO:

```
θ61° [isak.ohlsson@hasta:/home/proj/stage/housekeeper-bundles] [P_base] 6s $ bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-prod.sh -e P_main -t housekeeper -b master

...

repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/proj/stage/housekeeper-bundles
done.
Log deploy... done.
housekeeper, version 4.8.0
```

**Pull request merged**